chromium/browser_type.go

```diff
@@ -144,9 +144,10 @@ func (b *BrowserType) connect(
 }
 
 func (b *BrowserType) link(
+	ctx context.Context,
 	wsURL string, logger *log.Logger,
 ) (*common.BrowserProcess, error) {
-	bProcCtx, bProcCtxCancel := context.WithCancel(context.Background())
+	bProcCtx, bProcCtxCancel := context.WithCancel(ctx)
```

Comment:
It's good to get `Context` from outside for maintainability and testing. I'm trying to understand the whole fix, and I wonder if this change is related to it. I looked at [this](https://github.com/grafana/xk6-browser/blob/529532ec9371b0f283bca1b377c5481e7ad24f7c/browser/module.go#L106), and we pass `context.Background`.

Reply:
Initially i was wondering whether this could have caused the issue and reverted the changes from commits [1](https://github.com/grafana/xk6-browser/pull/1420/commits/e439161ccd7e32d1c00407970535c1362cf4d10d) and [2](https://github.com/grafana/xk6-browser/pull/1420/commits/c9734fe1a7f9350f9fdebb320d1c81fc14ac0dad). This didn't make a difference, but i left this in since it was better to pass the context in than to create a `context.Background`.

Reply:
It's OK. Just asked if it's related to the fix. It was not :)